datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -49,6 +53,62 @@ use super::coercion::{
     eq_coercion, like_coercion, numerical_coercion, order_coercion, string_coercion,
 };
 
+// Simple (low performance) kernels until optimized kernels are added to arrow
+// TODO: file arrow-rs ticket to track this feature
+
+fn is_distinct_from_bool(
+    left: &BooleanArray,
+    right: &BooleanArray,
+) -> Result<BooleanArray> {
+    // Different from `eq` beacause not eq_bool because null == null
+    Ok(left
+        .iter()
+        .zip(right.iter())
+        .map(|(left, right)| Some(left != right))
+        .collect())
+}
+
+fn is_not_distinct_from_bool(
+    left: &BooleanArray,
+    right: &BooleanArray,
+) -> Result<BooleanArray> {
+    Ok(left
+        .iter()
+        .zip(right.iter())
+        .map(|(left, right)| Some(left == right))
+        .collect())
+}
+
+#[allow(clippy::bool_comparison)]
```

Comment:
Clippy said `left < right` was clearer if expressed as `!left && right` which I happen to disagree with in this case, so I ignored the clippy lint

Reply:
I guess you can submit patch to them lol

Reply:
```suggestion
// TODO use arrow-rs kernels when available. See
// https://github.com/apache/arrow-rs/issues/959
#[allow(clippy::bool_comparison)]
```